x-pack/plugins/observability_solution/observability_logs_explorer/public/plugin.ts

```diff
@@ -123,7 +128,16 @@ export class ObservabilityLogsExplorerPlugin
     };
   }
 
-  public start(_core: CoreStart, _pluginsStart: ObservabilityLogsExplorerStartDeps) {
+  public start(core: CoreStart, _pluginsStart: ObservabilityLogsExplorerStartDeps) {
+    const { discover, fleet, logs } = core.application.capabilities;
+
+    if (!(discover?.show && fleet?.read && logs?.show)) {
+      this.appStateUpdater.next(() => ({
```

Comment:
Could you explain why going with the appStateUpdater instead of conditionally registering the app in the setup lifecycle?

Reply:
I see, I couldn't find any other registration like this, I can't suggest any better option. Let's stick to your initial approach 👌